packages/office-ui-fabric-react/src/components/TextField/TextField.types.ts

```diff
@@ -166,7 +168,7 @@ export interface ITextFieldProps extends React.AllHTMLAttributes<HTMLInputElemen
    * change the error message displayed (if any) based on the current value. `errorMessage` and
    * `onGetErrorMessage` are mutually exclusive (`errorMessage` takes precedence).
    */
-  errorMessage?: string;
+  errorMessage?: TextFieldErrorMessage;
```

Comment:
Since this is a very simple union type, I think it's better to just use `string | JSX.Element` rather than adding an extra typedef. (I talked to @jdhuntington about this separately and he's okay with it.) This is consistent with what most other places in our code do, is easier to understand at a glance, and avoids cluttering our exported API surface with extra types.